docker/Dockerfile-ca

```diff
@@ -1,47 +1,49 @@
-
-# Dependencies
-RUN apt-get -qq update
-RUN apt-get -qq  install curl software-properties-common ca-certificates gnupg -y
-RUN useradd -ms /bin/bash keybase
-USER keybase
+FROM alpine:latest AS builder
+
+# add dependencies
+RUN apk update && apk add --no-cache \
+    go curl git musl-dev
+
+# build keybase binary
+WORKDIR /go
+ENV GOPATH=/go
+ENV KEYBASE_VERSION=5.0.0
+RUN go get -d -v github.com/keybase/client/go/keybase
+RUN cd src/github.com/keybase/client/go/keybase && git checkout v$KEYBASE_VERSION
+RUN go install -tags production github.com/keybase/client/go/keybase
+
+# build kbfsfuse binary (we won't use FUSE but the bot needs KBFS for exchanging Team config files)
+WORKDIR /go/src/github.com/keybase/client/go/kbfs/kbfsfuse
+RUN go install 
+
+WORKDIR /temp
+COPY ./ /temp
+RUN go mod download
+RUN go build -o bin/keybaseca src/cmd/keybaseca/keybaseca.go
+
+FROM alpine:latest
+
+# add bash for entrypoint scripts, ssh for ssh-keygen used by the bot, sudo for stepping down to keybase user
+RUN apk update && apk add --no-cache bash openssh sudo
 WORKDIR /home/keybase
 
-# Download and verify the deb
-# Key fingerprint from https://keybase.io/docs/server_security/our_code_signing_key
-RUN curl --remote-name https://prerelease.keybase.io/keybase_amd64.deb
-RUN curl --remote-name https://prerelease.keybase.io/keybase_amd64.deb.sig
-# Import our gpg key from our website. Pulling from key servers caused a flakey build so
-# we get the key from the Keybase website instead.
-RUN curl -sSL https://keybase.io/docs/server_security/code_signing_key.asc | gpg --import
-# This line will error if the fingerprint of the key in the file does not match the
-# known fingerprint of the our PGP key
-RUN gpg --fingerprint 222B85B0F90BE2D24CFEB93F47484E50656D16C7
-# And then verify the signature now that we have the key
-RUN gpg --verify keybase_amd64.deb.sig keybase_amd64.deb
-
-# Silence the error from dpkg about failing to configure keybase since `apt-get install -f` fixes it
-USER root
-RUN dpkg -i keybase_amd64.deb || true
-RUN apt-get install -fy
-USER keybase
-
-# Install go
-USER root
-RUN add-apt-repository ppa:gophers/archive -y
-RUN apt-get update
-RUN apt-get install golang-1.11-go git sudo -y
-USER keybase
-
-# Install go dependencies (speeds up future builds)
-COPY --chown=keybase go.mod .
-COPY --chown=keybase go.sum .
-RUN /usr/lib/go-1.11/bin/go mod download
-
-COPY --chown=keybase ./ /home/keybase/
-
-RUN /usr/lib/go-1.11/bin/go build -o bin/keybaseca src/cmd/keybaseca/keybaseca.go
+# add the keybase user
+RUN adduser -s /bin/bash -h /home/keybase -D keybase
+
+# copy the keybase binaries from previous build step 
+COPY --from=builder /go/bin/keybase /usr/local/bin/
+COPY --from=builder /go/bin/kbfsfuse /usr/local/bin/
+COPY --from=builder /temp/bin/keybaseca /home/keybase/bin/keybaseca
+
+# copy in entrypoint scripts and fix permissions
+COPY ./docker/entrypoint-generate.sh .
+COPY ./docker/entrypoint-server.sh .
+COPY ./docker/env.sh .
```

Comment:
Adding this line means that `integrationTest.sh` now depends on the env.sh file existing in the docker directory (when this dependency didn't exist before). As a simple fix, can you add a `touch docker/env.sh` to `integrationTest.sh` so that the file is guaranteed to exist?

Reply:
the env vars shouldn't be built into the docker image. they should be passed into the `docker run` command